modules/platforms/cpp/ignite/protocol/utils.cpp

```diff
@@ -28,20 +28,73 @@
 
 namespace ignite::protocol {
 
+/**
+ * Check if int value fits in @c T.
+ *
+ * @tparam T Int type to fit value to.
+ * @param value Int value.
+ */
+template<typename T>
+inline void check_int_fits(std::int64_t value) {
```

Comment:
This looks suspicious. I think we can use MsgPack type code prefixes to check if the next value in the stream is `int` or something else.